src/smk/tool/geomark/tool-geomark.js

```diff
@@ -211,14 +211,45 @@ include.module( 'tool-geomark', [
             }
 
             this.toGeomark = function(geomarkInfo, drawingLayer) {
+                const geomarkProperties = this.getGeomarkProperties(geomarkInfo);
                 return {
-                    id: geomarkInfo.id || geomarkInfo.properties.id,
-                    url: geomarkInfo.url || geomarkInfo.properties.url,
+                    id: geomarkInfo.id || geomarkProperties.id,
```

Comment:
The getGeomarkProperties function has a path where it returns an empty object. If this happens, accessing a property of an empty object will throw an unhandled exception (eg. geomarkProperties.id and geomarkProperties.url).

Reply:
I returned the empty object because I know that attempting to access a property of an undefined object would throw an exception ... but if accessing a property on an empty object ALSO throws an exception, returning an empty object is no better. 

The only other thought I have is to throw an exception - how does that sound? I'll try it out, and move the call to showCouldNotLoadMessage() into the catch block. Same for the getGeometry() function, which can also return an empty object.

Reply:
No, you're right. The property values will just be undefined, I'm getting my languages mixed up! But is a geomark with an undefined id and/or url valid?